tests/cecil-tests/ApiCapitalizationTest.cs

```diff
@@ -0,0 +1,252 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Versioning;
+using System.Text;
+using System.Text.RegularExpressions;
+using NUnit.Framework;
+
+using Mono.Cecil;
+using Xamarin.Tests;
+using Xamarin.Utils;
+using System.Configuration.Assemblies;
+using MethodAttributes = Mono.Cecil.MethodAttributes;
+
+#nullable enable
+
+namespace Cecil.Tests {
+	[TestFixture]
+	public class ApiCapitalizationTest {
+
+		Dictionary<string, AssemblyDefinition> assemblyCache = new Dictionary<string, AssemblyDefinition> ();
+
+
+		[OneTimeSetUp]
+		public void SetUp ()
+		{
+			foreach (string assemblyPath in Helper.NetPlatformAssemblies) {
+
+				var assembly = Helper.GetAssembly (assemblyPath);
+				assemblyCache.Add (assemblyPath, assembly);
+			}
+		}
+
+		[OneTimeTearDown]
+		public void TearDown ()
+		{
+			assemblyCache.Clear ();
+		}
+
+		bool IsTypeObsolete (TypeDefinition type)
+		{
+			if (type == null)
+				return false;
+
+			if (type.HasCustomAttributes && type.CustomAttributes.Where ((t) => t.AttributeType.Name == "ObsoleteAttribute" || t.AttributeType.Name == "AdviceAttribute").Any ()) {
+				return true;
+			}
```

Comment:
This is no big deal, but we often prefer shorter lines. Long conditions can often be quite hard to read, so I think something like this would be easier on the eyes/brain:

```suggestion
			if (!type.HasCustomAttributes)
				return false;

			return type.CustomAttributes.Any ((t) => 
				t.AttributeType.Name == "ObsoleteAttribute" ||
				t.AttributeType.Name == "AdviceAttribute")
			);
```